lib/install/management/store_files.go

```diff
@@ -72,19 +71,44 @@ func (d *Dispatcher) DeleteStores(vchVM *vm.VirtualMachine, conf *config.Virtual
 	return nil
 }
 
-func (d *Dispatcher) deleteParent(ds *object.Datastore, root string) error {
-	defer trace.End(trace.Begin(""))
+func (d *Dispatcher) deleteImages(conf *config.VirtualContainerHostConfigSpec) error {
+	var emptyImages bool
+	var errs []string
+
+	for _, imageDir := range conf.ImageStores {
+		imageDSes, err := d.session.Finder.DatastoreList(d.ctx, imageDir.Host)
+		if err != nil {
+			errs = append(errs, err.Error())
+			continue
+		}
+
+		if len(imageDSes) != 1 {
+			errs = append(errs, fmt.Sprintf("Invalid or ambiguous datastore name %s provided while attempting to remove image stores", imageDir.Host))
+			continue
+		}
 
-	_, err := d.deleteDatastoreFiles(ds, root, true)
-	return err
+		if emptyImages, err = d.deleteDatastoreFiles(imageDSes[0], imageDir.Path, true); err != nil {
```

Comment:
I don't think this will work consistently on VSAN.  VSAN has complicated and inconsistent behavior when deleting a directory full of child directories with `vmdk` files.  You'll need to delete the children first before deleting the files in most cases or this will return an error.  @emlin has a helper to do this in `vic-machine` somewhere.